students/hirot_directory/session10/oo_mailroom_hiro/mailroom.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+
+import sys
+
+
+def request_donation():
+    donor_name = str(input("Full Name: "))
+    new_donation = int(input("How much do you want to donate?: "))
+    if (new_donation > 0):
+        print("Thank you for donation, Mr./Ms. {}! Your donation amount is USD ${}.".format(donor_name, new_donation))
+    return Donor(donor_name, new_donation)
+
+
+def select_submenu():
+    choice = int(input("Select from one of these options:\n"
+                       "(1) Show a list\n"
+                       "(2) Enter a donation\n"
+                       "(3) Exit\n"))
+    return choice
+
+
+def select_mainmenu():
+    answer = int(input("Select from one of these options:\n"
+                       "(1) Send a Thank you\n"
+                       "(2) Create a Report\n"
+                       "(3) Send letters to everyone\n"
+                       "(4) quit\n"))
+    return answer
+
+
+def select_quit():
+    selection = (input("Are you sure to quit? (Y or N) To return to main menu, please select 'N'.: ")).upper()
+    return selection
+
+
+def mainloop():
+    while True:
+        try:
+            answer = select_mainmenu()
+            dict_answer[answer]()
+
+        except ValueError as e:
+            print("Error happened. Error is ", type(e), "Please type 1, 2 or 3")
+        except KeyError as k:
+            print("Error happened. Error is ", type(k), "Please type 1, 2 or 3")
+
+
+def thank_you_loop():
+    while True:
+        try:
+            choice = select_submenu()
+            dict_choice[choice]()
+
+        except ValueError as e:
+            print("Error happened. Error is ", type(e) , "Please type 1, 2 or 3")
+        except KeyError as k:
+            print("Error happened. Error is ", type(k) , "Please type 1, 2 or 3")
+
+
+def quit():
+    print("quit function is called")
+
+    try:
+        selection = select_quit()
+        if (selection == 'Y'):
+            print("Good bye...")
+            sys.exit()
+        else:
+            return mainloop()
+
+    except ValueError as e:
+        print("Error happened. Error is ", type(e), "Please type 1, 2 or 3")
+    except KeyError as k:
+        print("Error happened. Error is ", type(k), "Please type 1, 2 or 3")
+
+
+def make_sample_db():
+
+    donor_names = ["William Gates III", "Mark Zuckerberg", "Jeff Bezos", "Paul Allen"]
+    donations = [[200, 500], [2000, 3000], [4000, 5000], [100, 150]]
+
+    sample_db = {}
+
+    for name, donation in zip(donor_names, donations):
+        sample_db[name] = donation
+
+    return sample_db
+
+    # return [Donor("William Gates III", [653772.32, 12.17]),
+    #         Donor("Jeff Bezos", [877.33]),
+    #         Donor("Paul Allen", [663.23, 43.87, 1.32]),
+    #         Donor("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),
+    #         ]
+
+
+def find_donor(name):
+    if name in make_sample_db().sample_db.keys():
```

Comment:
if name in make_sample_db().sample_db() would do the dame thing.

But this should be finding teh donor IN this DonorDB!!!

if name in self.all_donors:

should do it.

and you don't want to return the name, you want to return the donor!

to make it more compact:

`return self.all_donors.get(name, None)`

That will give you the donor associated with the name if it's there, and None otherwise.